omnibus/files/private-chef-ctl-commands/secrets.rb

```diff
@@ -74,9 +106,41 @@ def confirm_continue!(message)
   end
 end
 
-def set_secret_(group, key, secret)
+def set_secret_(group, key, secret, with_restart=nil)
   credentials.add(group, key, value: secret, frozen: true, force: true)
   credentials.save
+  lookup = "#{group}.#{key}"
+  puts "#{lookup} changed."
+
+  #TODO: do we need to guard against services being empty? (KNOWN_CREDS being out of sync w/ service list)
```

Comment:
The connection between`SERVICES_REQUIRING_RESTART` and `KNOWN_CREDS` is that *every key* of `SERVICES_REQUIRING_RESTART` should be in `KNOWN_CREDS`; but you're right, there could be a reason for adding something to `KNOWN_CREDS` that doesn't have a service depend on it.

```
irb(main):001:0> Array(nil)
=> []
```
So, I think what you have there is :+1: